astropy/wcs/wcsapi/high_level_api.py

```diff
@@ -117,165 +117,213 @@ def world_to_array_index(self, *world_objects):
+            w = world_objects[ikey]
+            if not isinstance(w, klass):
+                raise ValueError("Expected the following order of world "
+                                 "arguments: {}".format(', '.join([k.__name__ for (k, _, _) in classes.values()])))
+
+            # FIXME: For now SkyCoord won't auto-convert upon initialization
+            # https://github.com/astropy/astropy/issues/7689
+            from astropy.coordinates import SkyCoord
+            if isinstance(w, SkyCoord):
+                if 'frame' in kwargs:
+                    objects[key] = w.transform_to(kwargs['frame'])
                 else:
-                    objects[key] = klass_gen(world_by_key[key], *args, **kwargs)
+                    objects[key] = w
+            else:
+                objects[key] = klass_gen(w, *args, **kwargs)
 
+    # We now extract the attributes needed for the world values
+    world = []
+    for key, _, attr in components:
+        if callable(attr):
+            world.append(attr(objects[key]))
         else:
+            world.append(rec_getattr(objects[key], attr))
 
-            for ikey, key in enumerate(classes):
+    return world
 
-                klass, args, kwargs, *rest = classes[key]
 
-                if len(rest) == 0:
-                    klass_gen = klass
-                elif len(rest) == 1:
-                    klass_gen = rest[0]
-                else:
-                    raise ValueError("Tuples in world_axis_object_classes should have length 3 or 4")
-
-                w = world_objects[ikey]
-                if not isinstance(w, klass):
-                    raise ValueError("Expected the following order of world "
-                                     "arguments: {}".format(', '.join([k.__name__ for (k, _, _) in classes.values()])))
-
-                # FIXME: For now SkyCoord won't auto-convert upon initialization
-                # https://github.com/astropy/astropy/issues/7689
-                from astropy.coordinates import SkyCoord
-                if isinstance(w, SkyCoord):
-                    if 'frame' in kwargs:
-                        objects[key] = w.transform_to(kwargs['frame'])
-                    else:
-                        objects[key] = w
-                else:
-                    objects[key] = klass_gen(w, *args, **kwargs)
+def values_to_high_level_objects(*world_values, low_level_wcs):
+    """
+    Convert low level values into high level objects.
 
-        # We now extract the attributes needed for the world values
-        world = []
-        for key, _, attr in components:
-            if callable(attr):
-                world.append(attr(objects[key]))
-            else:
-                world.append(rec_getattr(objects[key], attr))
+    This function uses the information in ``wcs.world_axis_object_classes`` and
+    ``wcs.world_axis_object_components`` to convert the high level objects
+    (such as `~.SkyCoord`) to low level "values" `~.Quantity` objects.
```

Comment:
```suggestion
    This function uses the information in ``wcs.world_axis_object_classes`` and
    ``wcs.world_axis_object_components`` to convert the low level "values"
    `~.Quantity` objects to.high level objects (such as `~.SkyCoord`).
```
this is the inverse to `high_level_objects_to_values`, right?

Reply:
Thanks!